llm_toolkit/models.py

```diff
@@ -198,7 +198,10 @@ def estimate_token_num(self, text) -> int:
     """Estimates the number of tokens in |text|."""
     # https://cookbook.openai.com/examples/how_to_count_tokens_with_tiktoken
     try:
-      encoder = tiktoken.encoding_for_model(self.name)
+      if 'gpt-4' in self.name:  # gpt-4 and gpt-4o
+        encoder = tiktoken.encoding_for_model('gpt-4')
```

Comment:
What is the reason for not specifying `gpt-4o` directly? https://github.com/openai/tiktoken/blob/c0ba74c238d18b4824c25f3c27fc8698055b9a76/tiktoken/model.py#L22-L23

Reply:
Hi, @DavidKorczynski. I see the link you provided claimed the support for `gpt-4o`. But it seems now it gets the error below if running the code `encoder = tiktoken.encoding_for_model(self.name)` for `gpt-4o` directly:
```
Python 3.11.7 (main, Dec  8 2023, 18:56:57) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tiktoken
>>> encoder = tiktoken.encoding_for_model("gpt-4o")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kaixuan/FDG_LLM/oss-fuzz-gen/.venv/lib/python3.11/site-packages/tiktoken/model.py", line 97, in encoding_for_model
    return get_encoding(encoding_name_for_model(model_name))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kaixuan/FDG_LLM/oss-fuzz-gen/.venv/lib/python3.11/site-packages/tiktoken/model.py", line 84, in encoding_name_for_model
    raise KeyError(
KeyError: 'Could not automatically map gpt-4o to a tokeniser. Please use `tiktoken.get_encoding` to explicitly get the tokeniser you expect.'
```
you can confirm it again on your env.

Reply:
Likely because of https://github.com/google/oss-fuzz-gen/blob/3d209143f94acf6a1863acdaed69c03734e1602c/requirements.in#L14 -- let's bump?

Reply:
Ok~ 
I have tried `tiktoken 0.7.0`, it works now if directly running `encoder = tiktoken.encoding_name_for_model("gpt-4o")` after upgrading the tiktoken.

Reply:
@DavidKorczynski Do I need to recreate a PR to directly use `tiktoken` to get the encoding for gpt-4o? Or we can continue use this PR (if you can revise the code)?

Reply:
Got it. I have finished it~